sdks/python/apache_beam/io/gcp/bigquery_tools.py

```diff
@@ -101,6 +101,10 @@
 # Timeout for a BQ streaming insert RPC. Set to a maximum of 2 minutes.
 BQ_STREAMING_INSERT_TIMEOUT_SEC = 120
 
+_PROJECT_PATTERN = r'([a-z0-9.-]+:)?[a-z][a-z0-9-]{4,28}[a-z0-9]'
```

Comment:
did these pattern come from some codebase or constructed manually?

Reply:
It's the regex we use for the AutoML Tables product. `([a-z0-9.-]+:)?` handles domain-scoped projects, and `[a-z][a-z0-9-]{4,28}[a-z0-9]` is the most straightforward way to satisfy https://cloud.google.com/resource-manager/docs/creating-managing-projects#before_you_begin:

- Must be 6 to 30 characters in length.
- Can only contain lowercase letters, numbers, and hyphens.
- Must start with a letter.
- Cannot end with a hyphen.

Reply:
I think I would refrain from imposing length restrictions on project id it may change in the future, but SDK will not. WDYT?

Reply:
sgtm, removed length restrictions for project id